packages/core/backend/sirius-components-core-graphql/src/main/java/org/eclipse/sirius/components/core/graphql/datafetchers/ViewerOmniboxCommandsDataFetcher.java

```diff
@@ -35,58 +35,53 @@
-import graphql.relay.DefaultPageInfo;
 import graphql.relay.Edge;
 import graphql.relay.Relay;
 import graphql.schema.DataFetchingEnvironment;
-import reactor.core.publisher.Mono;
 
 /**
  * Data fetcher for the field Viewer#omniboxCommands.
  *
  * @author gcoutable
  */
 @QueryDataFetcher(type = "Viewer", field = "omniboxCommands")
-public class ViewerOmniboxCommandsDataFetcher implements IDataFetcherWithFieldCoordinates<CompletableFuture<Connection<OmniboxCommand>>> {
+public class ViewerOmniboxCommandsDataFetcher implements IDataFetcherWithFieldCoordinates<Connection<OmniboxCommand>> {
 
     private static final String CONTEXT_ENTRIES_ARGUMENT = "contextEntries";
 
     private static final String QUERY_ARGUMENT = "query";
 
-    private final IEditingContextEventProcessorRegistry editingContextEventProcessorRegistry;
+    private final List<IOmniboxCommandProvider> omniboxCommandProviders;
+
+    private final List<IOmniboxCommandOrderer> omniboxCommandOrderers;
 
     private final ObjectMapper objectMapper;
 
-    public ViewerOmniboxCommandsDataFetcher(IEditingContextEventProcessorRegistry editingContextEventProcessorRegistry, ObjectMapper objectMapper) {
-        this.editingContextEventProcessorRegistry = Objects.requireNonNull(editingContextEventProcessorRegistry);
+    public ViewerOmniboxCommandsDataFetcher(ObjectMapper objectMapper, List<IOmniboxCommandProvider> omniboxCommandProviders, List<IOmniboxCommandOrderer> omniboxCommandOrderers) {
         this.objectMapper = Objects.requireNonNull(objectMapper);
+        this.omniboxCommandProviders = Objects.requireNonNull(omniboxCommandProviders);
+        this.omniboxCommandOrderers = Objects.requireNonNull(omniboxCommandOrderers);
     }
 
     @Override
-    public CompletableFuture<Connection<OmniboxCommand>> get(DataFetchingEnvironment environment) throws Exception {
+    public Connection<OmniboxCommand> get(DataFetchingEnvironment environment) throws Exception {
         Object argument = environment.getArgument(CONTEXT_ENTRIES_ARGUMENT);
         List<OmniboxContextEntry> omniboxContextEntries = this.objectMapper.convertValue(argument, new TypeReference<List<OmniboxContextEntry>>() { });
         String query = environment.getArgument(QUERY_ARGUMENT);
-        var input = new GetOmniboxCommandsInput(UUID.randomUUID(), query);
-
-        if (!omniboxContextEntries.isEmpty()) {
-            var omniboxContextEntry = omniboxContextEntries.get(0);
-            return this.editingContextEventProcessorRegistry.dispatchEvent(omniboxContextEntry.id(), input)
-                    .filter(GetOmniboxCommandsPayload.class::isInstance)
-                    .map(GetOmniboxCommandsPayload.class::cast)
-                    .map(this::toConnection)
-                    .switchIfEmpty(emptyMono())
-                    .toFuture();
-        }
-        return emptyMono().toFuture();
-    }
+        var input = new GetOmniboxCommandsInput(UUID.randomUUID(), omniboxContextEntries, query);
```

Comment:
This input and its matching payload are not needed anymore since it does not use the editing context anymore.